pkgs/top-level/all-packages.nix

```diff
@@ -4691,7 +4691,7 @@ let
   sdcc = callPackage ../development/compilers/sdcc { };
 
   smlnjBootstrap = callPackage ../development/compilers/smlnj/bootstrap.nix { };
-  smlnj = callPackage_i686 ../development/compilers/smlnj { };
+  smlnj = callPackage ../development/compilers/smlnj { };
```

Comment:
I'm not sure if that's possible.  Another approach might be to put a conditional in `all-packages.nix` like:

```
smlnj = if stdenv.isDarwin 
          then callPackage ../development/compilers/smlnj { }
          else callPackage_i686 ../development/compilers/smlnj { };
```